### ISSUE TITLE
Need a way to specify fragments on k3po http script

### ISSUE THREAD

**Issue by dpwspoon** [k3po/k3po#385]
When writing an HTTP script I would like to write a fragment.  There is no way to do this in the [grammar](https://github.com/k3po/k3po/blob/develop/lang/src/main/antlr4/org/kaazing/k3po/lang/parser/v2/Robot.g4)

Attaching it to the URL does not work:

![image](https://cloud.githubusercontent.com/assets/2014623/21207508/445727ca-c21d-11e6-9b96-80dd75905ebe.png)

![image](https://cloud.githubusercontent.com/assets/2014623/21207512/48faa450-c21d-11e6-8e69-e0b9d9a9d331.png)

I would accept, either setting them on the URL as above (`accept http://localhost:8000/resource#fragment`) or a new write/read command for HTTP such as:

```
accept http://localhost:8000/resource#fragment
accepted
connected

read method "GET"
read fragment "fragment"
read closed
```

**Comment by dpwspoon** [comment]
It probably makes sense to do 
`read fragment "fragment"` as it allows regex, variables, etc

**Comment by dpwspoon** [comment]
http fragment does not make to the wire so this does not make sense

**Comment by cmebarrow** [comment]
@dpwspoon David, fragments can be part of the location header value in a redirect according to the HTTP 1.1 spec, see http://httpwg.org/specs/rfc7231.html#header.location. Do we not need to deal with that?

**Comment by dpwspoon** [comment]
@cmebarrow No, not in the scripting language as that is already a part of what was allowed on headers with generic values, i.e:

```
write header "Location" "/People.html#tim"
read header "Location"  "http://www.example.net/index.html"

write header "Location" /some-fancy-regex/
read header "Location"  /some-fancy-regex/
```

Thanks for clarifying as that part of the spec was the reason I originally opened this, but I was mistakenly thinking fragments could make it onto the accept/connect URI which would conceptually mean they were on the request line of an HTTP request, i.e.

```
GET /path?query#fragment
```

which appears to not be the case, see [request-target syntax](https://tools.ietf.org/html/rfc7230#section-5.3)

**Comment by jitsni** [comment]
fragment doesn't go on the wire, they are handled by the browsers

**Comment by cmebarrow** [comment]
@dpwspoon OK thanks for explaining

**Comment by jfallows** [comment]
There is still the scenario of a server issuing a 302 redirect `Location` header with fragment included, captured as a variable expression, then used directly as a connect location.

In this case, we just need to make sure we are dropping the fragment from the path in the request.

**Comment by dpwspoon** [comment]
The tests for that are here: https://github.com/k3po/k3po/pull/386.  I'm not sure if they strictly enforce that that the fragment is not sent.  Reopening this ticket as an action item to confirm.
